types/applications/baseEntitySheet.d.ts

```diff
@@ -78,7 +78,7 @@ declare namespace BaseEntitySheet {
   interface Data<O extends Entity = Entity> extends FormApplication.Data<O> {
     cssClass: string
     editable: boolean
-    entity: EntityData<O extends Entity<infer D> ? D : never>
```

Comment:
Note to self: check later if that is indeed just `data` or `data.data`.

Reply:
FYI, the version I committed extracts Whatever Entity.data type is.